prerendering.bs

```diff
@@ -478,172 +459,146 @@ Every {{Document}} has an <dfn for="Document">activation start time</dfn>, which
 
-  1. If |resource| is a [=request=] and we [=can activate a prerender=] given |browsingContext|, |historyHandling|, <var ignore>navigationType</var>, and |resource|, then:
+  1. Let |matchingPrerenderedNavigable| be the result of [=getting the matching prerendering navigable=] given |navigable|, <var ignore>url</var>, <var ignore>cspNavigationType</var>, and <var ignore>documentResource</var>.
 
-    1. Let |successorBC| be |browsingContext|'s [=active document=]'s [=Document/prerendering browsing contexts map=][|resource|'s [=request/URL=]].
+  1. If |matchingPrerenderedNavigable| is not null, then:
 
-    1. Wait until all navigation attempts of |successorBC| have been [=navigate/mature|matured=].
+    1. Wait until either |matchingPrerenderedNavigable|'s [=navigable/active document=]'s [=Document/is initial about:blank=] is false, or |matchingPrerenderedNavigable| is [=destroy a top-level traversable|destroyed=].
 
-    1. If |successorBC| has not been [=discard|discarded=], then:
+    1. If |matchingPrerenderedNavigable| was not destroyed, then:
 
-      1. [=prerendering browsing context/Activate=] |successorBC| in place of |browsingContext| given |historyHandling|.
+      1. [=prerendering traversable/Activate=] |matchingPrerenderedNavigable| in place of |navigable| given <var ignore>historyHandling</var>.
 
-      1. Return.
+      1. Abort these steps.
 </div>
 
-Navigation redirects can also [=prerendering browsing context/activate=] [=prerendering browsing contexts=]. This is defined in the [[#navigate-fetch-patch]] section.
-
 <h3 id="navigate-fetch-patch">Navigation fetch changes</h3>
 
-<div algorithm="process a navigate fetch patch">
-  In [=process a navigate fetch=], append the following steps after the first sub-step under "While true:":
+<div algorithm="create navigation params by fetching patch">
+  In [=create navigation params by fetching=], add the following step near the top of the algorithm:
 
-  1. If we [=can activate a prerender=] given |browsingContext|, |historyHandling|, <var ignore>navigationType</var>, and <var ignore>request</var>, then:
+  1. Let |initiatorOrigin| be <var ignore>entry</var>'s [=session history entry/document state=]'s [=document state/initiator origin=].
 
-    1. [=prerendering browsing context/Activate=] |browsingContext| in place of <var ignore>sourceBrowsingContext</var> given |historyHandling|.
+  Append the following steps after the first sub-step under "While true:":
 
-    1. Return.
-
-  1. If |browsingContext| is a [=prerendering browsing context=]  and <var ignore>currentURL</var>'s [=url/origin=] is not [=/same site=] with |incumbentNavigationOrigin|, then:
+  1. If |navigable| is a [=prerendering navigable=] and <var ignore>currentURL</var>'s [=url/origin=] is not [=/same site=] with |initiatorOrigin|, then:
 
-    1. Run the [=environment discarding steps=] for <var ignore>navigationParams</var>'s [=navigation params/reserved environment=].
```

Comment:
Why is it no longer necessary to run the environment discarding steps?

Reply:
I think this step was incorrect in the first place. The environment discarding steps seem to only be used during redirects; overall failures to navigate, e.g. 204 or error documents, do not currently trigger these in HTML. I was thinking maybe they should? But that'd be a more general investigation.